File: 1221-split-a-string-in-balanced-strings/1221-split-a-string-in-balanced-strings.java
class Solution {
  public int balancedStringSplit(String s) {
    char[] charArray = s.toCharArray();
    int ans = 0;
    int count = 0;
    for (char c : charArray) {
      if (c == 'R') {
        count++;
      } else {
        count--;
      }
      if (count == 0) {
        ans++;
      }
    }
    return ans;
  }
}

File: 1221-split-a-string-in-balanced-strings/1221-split-a-string-in-balanced-strings.js
/**
 * @param {string} s
 * @return {number}
 */
var balancedStringSplit = function(s) {
    let ans = 0;
    let count = 0;
    for (let i = 0; i < s.length; i++) {
        let char = s.charAt(i);
        if (char === 'R') {
            count++;
        } else {
            count--;
        }

        if (count === 0) {
            ans++;
        }
    }
    return ans;
};
